apps/laboratory/src/components/Ethers/EthersSendCallsTest.tsx

```diff
@@ -0,0 +1,117 @@
+import { EthereumProvider } from '@walletconnect/ethereum-provider'
+import { useChakraToast } from '../Toast'
+import { parseGwei, type Address } from 'viem'
+import { vitalikEthAddress } from '../../utils/DataUtil'
+import { BrowserProvider } from 'ethers'
+import { EIP_5792_RPC_METHODS, getAtomicBatchSupportedChainInfo } from '../../utils/EIP5792Utils'
+
+export function EthersSendCallsTest() {
+  const toast = useChakraToast()
+  const { address, chainId, isConnected } = useWeb3ModalAccount()
+  const { walletProvider } = useWeb3ModalProvider()
+  const [loading, setLoading] = useState(false)
+  async function onSendCalls() {
+    try {
+      setLoading(true)
+      if (!walletProvider || !address) {
+        throw Error('user is disconnected')
+      }
+      if (!chainId) {
+        throw Error('chain not selected')
+      }
+      const provider = new BrowserProvider(walletProvider, chainId)
+      const amountToSend = parseGwei('0.001').toString(16)
+      const calls = [
+        {
+          to: vitalikEthAddress as `0x${string}`,
+          data: '0x' as `0x${string}`,
+          value: `0x${amountToSend}`
+        },
+        {
+          to: vitalikEthAddress as Address,
+          value: '0x00',
+          data: '0xdeadbeef'
+        }
+      ]
+      const sendCallsParams = {
+        version: '1.0',
+        chainId: `0x${BigInt(chainId).toString(16)}`,
+        from: address,
+        calls
+      }
+      const batchCallHash = await provider.send(EIP_5792_RPC_METHODS.WALLET_SEND_CALLS, [
+        sendCallsParams
+      ])
+      toast({
+        title: 'Success',
+        description: batchCallHash,
+        type: 'success'
+      })
+    } catch {
+      toast({
+        title: 'Error',
+        description: 'Failed to send calls',
+        type: 'error'
+      })
+    } finally {
+      setLoading(false)
+    }
+  }
+
+  function isSendCallsSupported(): boolean {
+    if (walletProvider instanceof EthereumProvider) {
+      return Boolean(
+        walletProvider?.signer?.session?.namespaces?.['eip155']?.methods?.includes(
+          EIP_5792_RPC_METHODS.WALLET_SEND_CALLS
+        )
+      )
+    }
+
+    return false
+  }
+
+  if (!isConnected || !walletProvider || !address) {
+    return (
+      <Text fontSize="md" color="yellow">
+        Wallet not connected
+      </Text>
+    )
+  }
+
+  if (!isSendCallsSupported()) {
+    return (
+      <Text fontSize="md" color="yellow">
+        Wallet does not support this feature
+      </Text>
+    )
+  }
+
+  const allowedChains =
+    walletProvider instanceof EthereumProvider
+      ? getAtomicBatchSupportedChainInfo(walletProvider, address)
+      : []
+
+  if (allowedChains.length === 0) {
+    return (
+      <Text fontSize="md" color="yellow">
+        Account does not support this feature
+      </Text>
+    )
+  }
+
+  return allowedChains.find(chainInfo => chainInfo.chainId === Number(chainId)) ? (
```

Comment:
Could we avoid inline array calls like this? 

```
const allowedChainNames = allowedChains.map(ci => ci.chainName).join(', ');
const currentChainInfo = allowedChains.find(chainInfo => chainInfo.chainId === Number(chainId));

...

return currentChainInfo ? (
  <Stack direction={['column', 'column', 'row']}>
    ...
  </Stack>
) : (
  <Text fontSize="md" color="yellow">
    Switch to {allowedChainNames} to test this feature
  </Text>
)
```

Reply:
Done